src/components/MultilevelSelect/index.js

```diff
@@ -1,50 +1,121 @@
-import { useCallback, useEffect, useState } from 'react';
+/* eslint-disable no-param-reassign */
+import { useCallback, useRef, useState } from 'react';
 import styled from 'styled-components';
-import { Col, Row } from 'antd';
 import PropTypes from 'prop-types';
+import { Button, Empty, Modal } from 'antd';
 
-import TreeViewer from './TreeViewer';
+import { useConfig } from 'contexts/config';
+
+import TreeViewer from 'components/TreeViewer';
+import { Tree } from 'entities/Tree';
 import SelectBoard from './SelectBoard';
 
-const Wrapper = styled(Row).attrs(props => ({
-  gutter: [8, 8],
-  ...props,
-}))`
-  padding: 1rem;
-  border: 1px solid black;
-  border-radius: 20px;
+const Header = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const Wrapper = styled.div`
+  display: flex;
 `;
 
-const TreeWrapper = styled(Col).attrs(props => ({
-  span: 10,
-  ...props,
-}))``;
+const TreeWrapper = styled.div`
+  min-width: 300px;
+  max-width: 500px;
+  margin-right: 1rem;
+`;
 
-const MultilevelSelect = ({ treeData }) => {
-  const [activeNode, setActiveNode] = useState({});
-  const fakeApi = () => {};
+const BoardWrapper = styled.div`
+  width: 100%;
+`;
+
+const StyledEmpty = styled(Empty)`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+`;
+const fakeApi = () => {};
+
+const MultilevelSelect = ({ defaultTreeData, open, onCancel, onFinish }) => {
+  const treeInsRef = useRef(new Tree(defaultTreeData));
```

Comment:
I think `const treeInsRef = useRef(new Tree(defaultTreeData)).current;` is enough